### ISSUE TITLE
[Time Field] Input does not have programaticaly associated label

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3058]
## Current behavior

`sl-time-field` uses `input` element that has `ID`. In code there is also `sl-label` with `label` element inside it. `label` has an empty `for` attribute that means it's not associated with any form control.
Screen reader users that places focus on time field input hear "PLACEHOLDER TEXT, edit text, blank" (when placeholder is presented) or "CURRENT INPUT VALUE, edit text" (when input was filled in).
There is no information about name of field that is presented in `label`.

<img width="167" height="138" alt="Image" src="https://github.com/user-attachments/assets/dd5f7c6f-df93-4dbc-bd44-d6b1097fe65d" />

<img width="509" height="270" alt="Image" src="https://github.com/user-attachments/assets/34df6927-fa51-4ea0-a5b9-b5653cf57f6b" />

## Proposed solution

It's recommended to add input's `ID` value as label's `for` attribute value. 
Screen readers would announce 'Time PLACEHOLDER TEXT, edit text' / 'CURRENT INPUT VALUE Time, edit text'

<img width="508" height="263" alt="Image" src="https://github.com/user-attachments/assets/05563bdc-de11-4e4b-8904-5ed3a35db122" />

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
